library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -208,11 +201,14 @@ private void init(@Nullable AttributeSet attributeSet) {
         currentLocationBtn.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View v) {
-                focusOnUserLocation(true);
-
-                // Enable asking for enabling the location by resetting this flag in case it was true
-                hasAlreadyRequestedEnableLocation = false;
-                setWarmGps(true);
+                    // Enable asking for enabling the location by resetting this flag in case it was true
+                    hasAlreadyRequestedEnableLocation = false;
+                    setWarmGps(true, null, null, new OnLocationServicesEnabledCallBack() {
+                        @Override
+                        public void onSuccess() {
+                            focusOnUserLocation(true, locationBufferRadius);
```

Comment:
This does handle cases where location were disabled and user accepted to enable them.

Reply:
Sorry, I don't understand what this means, could you rephrase?

Reply:
@githengi how do you suggest this is handled?

`status.startResolutionForResult(activity, Constants.RequestCode.LOCATION_SETTINGS);` this is what checks the settings and returns it's result in an activity.

I guess a bigger question is whether clicking the button again after enabling significantly affect UX?

Reply:
Yes that's is not a  good UX, having to click my location again, users may not be able to easily know that they have to click it again.

Why not call `focusOnUserLocation(true, locationBufferRadius);` before or `setWarmGps` so that `onResume` takes user to their location.

This was the logic before as on master

Reply:
The whole point behind the callback approach is not to call `focusOnUserLocation(true, locationBufferRadius)` before.

I have added some logic in onResume that should help with this.